File: frontend/src/app/pages/third-parties-page/third-party.ts
export interface ThirdParty {
    id: number | null;
    firstName: string;
    lastName: string;
    birthDate: string;
    address: string;
    postalCode: string;
    city: string;
}

export interface ExistingThirdParty extends ThirdParty {
    id: number;
}

File: backend/src/main/java/fr/iut/valence/but/info/ctvr/backend/model/ThirdParty.java
package fr.iut.valence.but.info.ctvr.backend.model;

import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDate;

@Data
@Table(name = "third_parties")
@Entity
public class ThirdParty {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 50)
    private String firstName;

    @Column(nullable = false, length = 50)
    private String lastName;

    @Column(nullable = false)
    private LocalDate birthDate;

    @Column(nullable = false, length = 100)
    private String address;

    @Column(nullable = false, columnDefinition = "CHAR(5)")
    private String postalCode;

    @Column(nullable = false, length = 50)
    private String city;
}
